proxy.h

```diff
@@ -272,13 +294,22 @@ using dependent_t = typename dependent_traits<T, Us...>::type;
 
 }  // namespace details
 
+template <class F>
+concept basic_facade = requires {
+  typename F::dispatch_types;
+  typename std::integral_constant<
```

Comment:
Looks like another dance on `std::integral_constant`. Does the standard provide a more specific way to check if a value can be evaluated during compile-time?

Reply:
I do not find any